src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3948,4 +3948,100 @@ public static void updatePinsIsRouted(Net net) {
+                if(n.getNet().equals(gnd) || n.getNet().equals(vcc)) continue; // Static nets do not have physical equivalents
+                createPhysNetFromLogical(d, n);
+            }
+        }
+
+    /**
+     * Creates a physical nets from the provided logical EDIF net. Assumes that the EDIF netlist is finalized.
+     * @param d The design that contains the complete logical netlist and where the physical net will be created.
+     * @param edifNet The EDIF Net to create into a physical net.
+     * @return A list of the SitePinInsts added to the newly created physical net
+     */
+    static public List<SitePinInst> createPhysNetFromLogical(Design d, EDIFHierNet edifNet) {
+        //check whether net already exists
+        if (d.getNet(edifNet.getHierarchicalNetName())  != null) return null;
+
+        Net net = d.createNet(edifNet); //create physical net
+
+        // Get source EDIF port inst
+        EDIFHierPortInst srcPort = null;
+        for (EDIFHierPortInst hierPortInst : d.getNetlist().getPhysicalPins(edifNet)) {
+            EDIFPortInst portInst = hierPortInst.getPortInst();
+            if (portInst.isOutput() && portInst.getCellInst() != null) {
+                srcPort = hierPortInst;
+                break;
+            }
+        }
+
+        // Get cell connected to port inst
+        Cell srcCell = d.getCell(srcPort.getFullHierarchicalInstName());
+
+        // Connect physical net to the physical cell pins corresponding to the logical ports
+        List<SitePinInst> sitePins = new ArrayList<>();
+        for(EDIFHierPortInst hierPortInst: d.getNetlist().getPhysicalPins(edifNet)) {
+            EDIFPortInst portInst = hierPortInst.getPortInst();
+            if (portInst.equals(srcPort)) continue;
+
+            EDIFCellInst cellInst = portInst.getCellInst();
+            Cell cell = cellInst == null ? null : d.getCell(cellInst.getName());
+            if (cell == null) continue;
+
+            // Only create connection if the net goes outside the site. This prevents sitePins instances from
+            // erroneously being created.
+            if(!isIntraSiteConnection(srcCell, srcPort.getPortInst(), cell, portInst))
+                sitePins.add(net.connect(cell, portInst.getName()));
+        }
+
+        if (srcCell != null && sitePins.size() > 0) {
+            net.connect(srcCell, srcPort.getPortInst().getName());
+        }
+
+        return sitePins;
+    }
+
+    /**
+     * checks if the two provided ports of two cells can be connected without leaving the site.
+     * @param srcCell source cell
+     * @param srcPort port of source Cell
+     * @param dstCell destination cell
+     * @param dstPort port of destination cell
+     * @return true if the ports can be connected without leaving the site. False otherwise.
+     */
+    private static boolean  isIntraSiteConnection(Cell srcCell, EDIFPortInst srcPort, Cell dstCell, EDIFPortInst dstPort) {
```

Comment:
```suggestion
    public static boolean isIntraSiteConnection(Cell srcCell, EDIFPortInst srcPort, Cell dstCell, EDIFPortInst dstPort) {
```
Seems like this might be more broadly useful